*: add release notes for 6.1.6 (#13170)
pingcap/docs: releases/release-6.1.6.md

---
title: TiDB 6.1.6 Release Notes
summary: Learn about the compatibility changes, improvements, and bug fixes in TiDB 6.1.6.
---

# TiDB 6.1.6 Release Notes

Release date: xxxx, 2023

TiDB version: 6.1.6

Quick access: [Quick start](https://docs.pingcap.com/tidb/v6.1/quick-start-with-tidb) | [Production deployment](https://docs.pingcap.com/tidb/v6.1/production-deployment-using-tiup) | [Installation packages](https://www.pingcap.com/download/?version=v6.1.6#version-list)

## Compatibility changes

- Compatibility change 1

## Improvements

+ TiDB

>     - Prepare Plan Cache 可以缓存 BatchPointGet 计划 [#42125](https://github.com/pingcap/tidb/issues/42125) @[qw4990](https://github.com/qw4990)

Author: ```suggestion
    - Supports caching the execution plan for `BatchPointGet` in Prepared Plan Cache [#42125](https://github.com/pingcap/tidb/issues/42125) @[qw4990](https://github.com/qw4990)
```

    (dup: release-5.1.5.md > Bug fixes> TiDB)- Fix the issue that range partitions allow multiple `MAXVALUE` partitions [#36329](https://github.com/pingcap/tidb/issues/36329)
    (dup: release-6.5.1.md > Bug fixes> TiDB)- Fix the issue that Plan Cache might cache Shuffle operators and return incorrect results [#38335](https://github.com/pingcap/tidb/issues/38335) @[qw4990](https://github.com/qw4990) @[qw4990](https://github.com/qw4990)
    (dup: release-6.5.1.md > Bug fixes> TiDB)- Fix the issue that data race in time zone might cause data-index inconsistency [#40710](https://github.com/pingcap/tidb/issues/40710) @[wjhuang2016](https://github.com/wjhuang2016) @[wjhuang2016](https://github.com/wjhuang2016)
    (dup: release-6.5.1.md > Bug fixes> TiDB)- Fix the issue that goroutine leak might occur in `indexMerge` [#41545](https://github.com/pingcap/tidb/issues/41545) [#41605](https://github.com/pingcap/tidb/issues/41605) @[guo-shaoge](https://github.com/guo-shaoge) @[guo-shaoge](https://github.com/guo-shaoge)

>     - 修复在使用 Cursor Fetch 且在 Execute、Fetch、Close 之间运行其它语句后，Fetch 与 Close 命令可能会返回错误结果或造成 TiDB Panic 的问题 [#40094](https://github.com/pingcap/tidb/issues/40094) [@YangKeao](https://github.com/YangKeao)

Author: ```suggestion
    - Fix the issue that, when using Cursor Fetch and running other statements among Execute, Fetch, and Close, the Fetch and Close commands might return incorrect results or cause TiDB to panic [#40094](https://github.com/pingcap/tidb/issues/40094) [@YangKeao](https://github.com/YangKeao)
```

>     - 修复了 DDL 在修改浮点类型，保持长度不变且减少小数位时旧数据仍然保持原样的问题 [#41281](https://github.com/pingcap/tidb/issues/41281) [@zimulala](https://github.com/zimulala)

Author: ```suggestion
    - Fix the issue that when modifying the floating-point type using DDL to keep the length unchanged and reduce the decimal places, the old data still remains the same [#41281](https://github.com/pingcap/tidb/issues/41281) [@zimulala](https://github.com/zimulala)
```

>     - 修复了 Join `information_schema.columns` 表会造成 TiDB panic 的问题 [#32459](https://github.com/pingcap/tidb/issues/32459) [@tangenta](https://github.com/tangenta)

Author: ```suggestion
    - Fix the issue that joining the `information_schema.columns` table causes TiDB to panic [#32459](https://github.com/pingcap/tidb/issues/32459) [@tangenta](https://github.com/tangenta)
```

>     - 修复了生成执行计划过程中，因为获取的 Info Schema 不一致而导致的 TiDB panic [#41622](https://github.com/pingcap/tidb/issues/41622) [@tiancaiamao](https://github.com/tiancaiamao)

Author: ```suggestion
    - Fix the issue that TiDB panic occurs due to inconsistent Info Schema being obtained when generating the execution plan [#41622](https://github.com/pingcap/tidb/issues/41622) [@tiancaiamao](https://github.com/tiancaiamao)
```

    (dup: release-7.0.0.md > Bug fixes> TiDB)- Fix the issue that TiFlash reports an error for generated columns during execution [#40663](https://github.com/pingcap/tidb/issues/40663) @[guo-shaoge](https://github.com/guo-shaoge) @[guo-shaoge](https://github.com/guo-shaoge)
    (dup: release-7.0.0.md > Bug fixes> TiDB)- Fix the issue that TiDB might produce incorrect results when different partitioned tables appear in a single SQL statement [#42135](https://github.com/pingcap/tidb/issues/42135) @[mjonss](https://github.com/mjonss) @[mjonss](https://github.com/mjonss)
    (dup: release-6.5.1.md > Bug fixes> TiDB)- Fix the issue that Plan Cache might cache Shuffle operators and return incorrect results [#38335](https://github.com/pingcap/tidb/issues/38335) @[qw4990](https://github.com/qw4990) @[fzzf678](https://github.com/fzzf678)

>     - 修复使用 Index Merge 的方式读取包含 SET 类型的列的表时，结果可能出错的问题 [#41293](https://github.com/pingcap/tidb/issues/41293) @[time-and-fate](https://github.com/time-and-fate)

Author: ```suggestion
    - Fix the issue that using Index Merge to read a table containing the `SET` type column might lead to incorrect results [#41293](https://github.com/pingcap/tidb/issues/41293) @[time-and-fate](https://github.com/time-and-fate)
```

    (dup: release-7.0.0.md > Bug fixes> TiDB)- Fix the issue that full index scans might cause errors when prepared plan cache is enabled [#42150](https://github.com/pingcap/tidb/issues/42150) @[fzzf678](https://github.com/fzzf678) @[fzzf678](https://github.com/fzzf678)

>     - 修复当 DDL 执行过程中，使用 PointGet 读取表的 SQL 可能会在执行时抛出 panic 的问题 [#41622](https://github.com/pingcap/tidb/issues/41622) @[tiancaiamao](https://github.com/tiancaiamao)

Author: ```suggestion
    - Fix the issue that SQL statements using `PointGet` to read a table during the execution of a DDL statement might throw a panic [#41622](https://github.com/pingcap/tidb/issues/41622) @[tiancaiamao](https://github.com/tiancaiamao)
```

>     - 修复事务内点更新之后，select 结果不正确的问题 [#28011](https://github.com/pingcap/tidb/issues/28011) @[zyguan](https://github.com/zyguan)

Author: ```suggestion
    - Fix the issue that after executing `PointUpdate` within a transaction, TiDB returns incorrect results for the `SELECT` statement [#28011](https://github.com/pingcap/tidb/issues/28011) @[zyguan](https://github.com/zyguan)
```

+ PD

    - (dup): release-6.5.1.md > Bug 修复> PD - 修复 Region Scatter 接口可能导致 leader 分布不均匀的问题 [#6017](https://github.com/tikv/pd/issues/6017) @[HunDunDM](https://github.com/HunDunDM)
    - 修复 Online recovery 超时不生效的问题 @[v01dstar](https://github.com/v01dstar)

+ TiFlash

    - (dup): release-6.6.0.md > Bug 修复> TiFlash - 修复半连接在计算笛卡尔积时，使用内存过量的问题 [#6730](https://github.com/pingcap/tiflash/issues/6730) @[gengliqi](https://github.com/gengliqi)
    - (dup): release-6.5.1.md > Bug 修复> TiFlash - 修复 TiFlash 日志搜索过慢的问题 [#6829](https://github.com/pingcap/tiflash/issues/6829) @[hehechen](https://github.com/hehechen)
    - 修复 `CI Collation` 下聚合函数结果不正确的问题 [#7002](https://github.com/pingcap/tiflash/pull/7002) @[xzhangxian1008](https://github.com/xzhangxian1008)
    - 修复 `Decimal Cast` 结果不正确的问题 [#7026](https://github.com/pingcap/tiflash/pull/7026) @[windtalker](https://github.com/windtalker)
    - 修复不能识别生成列的问题 [#6801](https://github.com/pingcap/tiflash/issues/6801) @[guo-shaoge](https://github.com/guo-shaoge)
    - 修复 `Decimal` 的舍入问题 [#7022](https://github.com/pingcap/tiflash/issues/7022) @[LittleFall](https://github.com/LittleFall)

+ Tools

    + Backup & Restore (BR)

    + TiCDC

>         - 修复同步数据时由于 `Update` 和 `Insert` 语句乱序可能导致的 `Duplicate entry` 错误问题 [#8635](https://github.com/pingcap/tiflow/pull/8635) @[sdojjy](https://github.com/sojjy)
>         - 修复 `float32` 类型解析错误导致的浮点数精度丢失问题 [#8519](https://github.com/pingcap/tiflow/pull/8519) @[3AceShowHand](https://github.com/3AceShowHand)

Reviewer: ```suggestion
        - 修复同步数据时由于 `Update` 和 `Insert` 语句乱序可能导致 `Duplicate entry` 错误的问题 [#8597](https://github.com/pingcap/tiflow/issues/8597) @[sdojjy](https://github.com/sojjy)
        - 修复 `float32` 类型解析错误导致浮点数精度丢失的问题 [#8490](https://github.com/pingcap/tiflow/issues/8490) @[3AceShowHand](https://github.com/3AceShowHand)
```

>         - 修复 `float32` 类型解析错误导致浮点数精度丢失的问题 [#8490](https://github.com/pingcap/tiflow/issues/8490) @[3AceShowHand](https://github.com/3AceShowHand)

Author: ```suggestion
        (dup: release-7.0.0.md > # Behavior changes)* TiCDC fixes the issue of incorrect encoding of `FLOAT` data in Avro [#8490](https://github.com/pingcap/tiflow/issues/8490) @[3AceShowHand](https://github.com/3AceShowHand)
```

Reviewer: 这一条是否要挪到兼容性更改里呢

>         - 修复由 PD 和 TiCDC 之间网络隔离引起的 TiCDC 程序异常退出的问题 [#8462](https://github.com/pingcap/tiflow/pull/8642) @[overvenus](https://github.com/overvenus)

Reviewer: ```suggestion
        - 修复由于 PD 和 TiCDC 之间的网络隔离引起 TiCDC 程序异常退出的问题 [#8562](https://github.com/pingcap/tiflow/issues/8562) @[overvenus](https://github.com/overvenus)
```

> + TiKV

Author: ```suggestion

+ TiKV
```

>         - 修复在同步没有主键且包含非空唯一索引的表时，数据可能丢失的问题 [#8420](https://github.com/pingcap/tiflow/issues/8420) @[zhaoxinyu](https://github.com/zhaoxinyu)

Author: ```suggestion
        (dup: release-6.5.1.md > Bug fixes> Tools> TiCDC)- Fix the data inconsistency that occurs when replicating data to a TiDB or MySQL sink and when `CHARACTER SET` is specified on the column that has the non-null unique index without a primary key [#8420](https://github.com/pingcap/tiflow/issues/8420) @[zhaoxinyu](https://github.com/zhaoxinyu)
```

>         - 修复在同步没有主键且包含非空唯一索引的表时，可能导致的数据丢失问题 [#8420](https://github.com/pingcap/tiflow/issues/8420) @[zhaoxinyu](https://github.com/zhaoxinyu)
>         - 修复 `db sorter` 使用内存没有正确受到 `cgroup memory limit` 限制的问题 [#8623](https://github.com/pingcap/tiflow/pull/8623) @[amyangfei](https://github.com/amyangfei)
>         - 优化 `cdc cli` 在遇到非法输入时的错误提示。 [#8442](https://github.com/pingcap/tiflow/pull/8442) @[CharlesCheung96](https://github.com/CharlesCheung96)

Reviewer: ```suggestion
        - 修复在同步没有主键且包含非空唯一索引的表时，数据可能丢失的问题 [#8420](https://github.com/pingcap/tiflow/issues/8420) @[zhaoxinyu](https://github.com/zhaoxinyu)
        - 修复 `db sorter` 使用内存时未受 `cgroup memory limit` 限制的问题 [#8588](https://github.com/pingcap/tiflow/issues/8588) @[amyangfei](https://github.com/amyangfei)
        - 优化 `cdc cli` 在遇到非法输入时的错误提示 [#7903](https://github.com/pingcap/tiflow/issues/7903) @[CharlesCheung96](https://github.com/CharlesCheung96)
```

>     - 修复 CPU 配额限制问题 [13084](https://github.com/tikv/tikv/issues/13084) @[BornChanger](https://github.com/BornChanger)
>     - 修复 snapshot last index 不正确的问题 [12618](https://github.com/tikv/tikv/issues/12618) @[LintianShi](https://github.com/LintianShi)

Author: ```suggestion
    - Fix the issue of CPU quota limitation [13084](https://github.com/tikv/tikv/issues/13084) @[BornChanger](https://github.com/BornChanger)
    - Fix the issue of incorrect snapshot last index [12618](https://github.com/tikv/tikv/issues/12618) @[LintianShi](https://github.com/LintianShi)

```

>     (dup: release-6.5.1.md > Bug fixes> PD)- Fix the issue that the Region Scatter might cause uneven distribution of leader [#6017](https://github.com/tikv/pd/issues/6017) @[HunDunDM](https://github.com/HunDunDM) @[HunDunDM](https://github.com/HunDunDM)

Reviewer 2: ```suggestion
    (dup: release-6.5.1.md > Bug fixes> PD)- Fix the issue that the Region Scatter might cause uneven distribution of leader [#6017](https://github.com/tikv/pd/issues/6017) @[HunDunDM](https://github.com/HunDunDM)
```

>     - 修复 Online recovery 超时不生效的问题 @[v01dstar](https://github.com/v01dstar)

Reviewer 2: ```suggestion
    - Fix the issue that the timeout mechanism of Online Unsafe Recovery does not work [#6107](https://github.com/tikv/pd/issues/6107) @[v01dstar](https://github.com/v01dstar)
```

>     (dup: release-6.6.0.md > Bug fixes> TiFlash)- Fix the issue that semi-joins use excessive memory when calculating Cartesian products [#6730](https://github.com/pingcap/tiflash/issues/6730) @[gengliqi](https://github.com/gengliqi) @[gengliqi](https://github.com/gengliqi)
>     (dup: release-6.5.1.md > Bug fixes> TiFlash)- Fix the issue that TiFlash log search is too slow [#6829](https://github.com/pingcap/tiflash/issues/6829) @[hehechen](https://github.com/hehechen) @[hehechen](https://github.com/hehechen)

Reviewer 2: ```suggestion
    (dup: release-6.6.0.md > Bug fixes> TiFlash)- Fix the issue that semi-joins use excessive memory when calculating Cartesian products [#6730](https://github.com/pingcap/tiflash/issues/6730) @[gengliqi](https://github.com/gengliqi)
    (dup: release-6.5.1.md > Bug fixes> TiFlash)- Fix the issue that TiFlash log search is too slow [#6829](https://github.com/pingcap/tiflash/issues/6829) @[hehechen](https://github.com/hehechen)
```

>     - 修复 `CI Collation` 下聚合函数结果不正确的问题 [#7002](https://github.com/pingcap/tiflash/pull/7002) @[xzhangxian1008](https://github.com/xzhangxian1008)
>     - 修复 `Decimal Cast` 结果不正确的问题 [#7026](https://github.com/pingcap/tiflash/pull/7026) @[windtalker](https://github.com/windtalker)
>     - 修复不能识别生成列的问题 [#6801](https://github.com/pingcap/tiflash/issues/6801) @[guo-shaoge](https://github.com/guo-shaoge)

Reviewer 2: ```suggestion
    (dup: release-7.0.0.md > Bug fixes> TiFlash)- Fix the issue that TopN/Sort operators produce incorrect results after enabling the new collation [#6807](https://github.com/pingcap/tiflash/issues/6807) @[xzhangxian1008](https://github.com/xzhangxian1008)
    (dup: release-7.0.0.md > Bug fixes> TiFlash)- Fix the issue that Decimal cast rounds up incorrectly in certain cases [#6994](https://github.com/pingcap/tiflash/issues/6994) @[windtalker](https://github.com/windtalker)
    - Fix the issue that TiFlash cannot recognize generated columns [#6801](https://github.com/pingcap/tiflash/issues/6801) @[guo-shaoge](https://github.com/guo-shaoge)
```

>     (dup: release-7.0.0.md > Bug fixes> TiFlash)- Fix the issue that Decimal division does not round up the last digit in certain cases [#7022](https://github.com/pingcap/tiflash/issues/7022) @[LittleFall](https://github.com/LittleFall) @[LittleFall](https://github.com/LittleFall)

Reviewer 2: ```suggestion
    (dup: release-7.0.0.md > Bug fixes> TiFlash)- Fix the issue that Decimal division does not round up the last digit in certain cases [#7022](https://github.com/pingcap/tiflash/issues/7022) @[LittleFall](https://github.com/LittleFall)
```

    (dup: release-6.5.1.md > Bug fixes> TiDB)- Fix the issue that goroutine leak might occur in `indexMerge` [#41545](https://github.com/pingcap/tidb/issues/41545) [#41605](https://github.com/pingcap/tidb/issues/41605) @[guo-shaoge](https://github.com/guo-shaoge) @[guo-shaoge](https://github.com/guo-shaoge)
    - Fix the issue that, when using Cursor Fetch and running other statements among Execute, Fetch, and Close, the Fetch and Close commands might return incorrect results or cause TiDB to panic [#40094](https://github.com/pingcap/tidb/issues/40094) [@YangKeao](https://github.com/YangKeao)
    - Fix the issue that when modifying the floating-point type using DDL to keep the length unchanged and reduce the decimal places, the old data still remains the same [#41281](https://github.com/pingcap/tidb/issues/41281) [@zimulala](https://github.com/zimulala)
    - Fix the issue that joining the `information_schema.columns` table causes TiDB to panic [#32459](https://github.com/pingcap/tidb/issues/32459) [@tangenta](https://github.com/tangenta)

>     - Fix the issue that TiDB panic occurs due to inconsistent Info Schema being obtained when generating the execution plan [#41622](https://github.com/pingcap/tidb/issues/41622) [@tiancaiamao](https://github.com/tiancaiamao)

Reviewer 2: ```suggestion
    - Fix the issue that TiDB panic occurs due to inconsistent InfoSchema being obtained when generating the execution plan [#41622](https://github.com/pingcap/tidb/issues/41622) [@tiancaiamao](https://github.com/tiancaiamao)
```

---
title: TiDB 6.1.6 Release Notes
summary: Learn about the compatibility changes, improvements, and bug fixes in TiDB 6.1.6.
---

# TiDB 6.1.6 Release Notes

Release date: xxxx, 2023

TiDB version: 6.1.6

Quick access: [Quick start](https://docs.pingcap.com/tidb/v6.1/quick-start-with-tidb) | [Production deployment](https://docs.pingcap.com/tidb/v6.1/production-deployment-using-tiup) | [Installation packages](https://www.pingcap.com/download/?version=v6.1.6#version-list)

## Compatibility changes

> - Compatibility change 1

Author: ```suggestion
- TiCDC fixes the issue of incorrect encoding of FLOAT data in Avro [#8490](https://github.com/pingcap/tiflow/issues/8490) @[3AceShowHand](https://github.com/3AceShowHand)

    When upgrading the TiCDC cluster to v7.0.0, if a table replicated using Avro contains the `FLOAT` data type, you need to manually adjust the compatibility policy of Confluent Schema Registry to `None` before upgrading so that the changefeed can successfully update the schema. Otherwise, after upgrading, the changefeed will be unable to update the schema and enter an error state.
```

    (dup: release-6.5.1.md > Bug fixes> TiFlash)- Fix the issue that TiFlash log search is too slow [#6829](https://github.com/pingcap/tiflash/issues/6829) @[hehechen](https://github.com/hehechen)
    (dup: release-7.0.0.md > Bug fixes> TiFlash)- Fix the issue that TopN/Sort operators produce incorrect results after enabling the new collation [#6807](https://github.com/pingcap/tiflash/issues/6807) @[xzhangxian1008](https://github.com/xzhangxian1008)
    (dup: release-7.0.0.md > Bug fixes> TiFlash)- Fix the issue that Decimal cast rounds up incorrectly in certain cases [#6994](https://github.com/pingcap/tiflash/issues/6994) @[windtalker](https://github.com/windtalker)
    - Fix the issue that TiFlash cannot recognize generated columns [#6801](https://github.com/pingcap/tiflash/issues/6801) @[guo-shaoge](https://github.com/guo-shaoge)
    (dup: release-7.0.0.md > Bug fixes> TiFlash)- Fix the issue that Decimal division does not round up the last digit in certain cases [#7022](https://github.com/pingcap/tiflash/issues/7022) @[LittleFall](https://github.com/LittleFall)

+ Tools

    + Backup & Restore (BR)

>     + TiCDC

Reviewer: ```suggestion
    + TiCDC
 
```

>         - 修复同步数据时由于 `Update` 和 `Insert` 语句乱序可能导致 `Duplicate entry` 错误的问题 [#8597](https://github.com/pingcap/tiflow/issues/8597) @[sdojjy](https://github.com/sojjy)

Reviewer: ```suggestion
        - Fix the issue that the `Duplicate entry` error might occur during data replication due to the disorder of `Update` and `Insert` statements [#8597](https://github.com/pingcap/tiflow/issues/8597) @[sdojjy](https://github.com/sojjy)
```

        - 修复 `float32` 类型解析错误导致浮点数精度丢失的问题 [#8490](https://github.com/pingcap/tiflow/issues/8490) @[3AceShowHand](https://github.com/3AceShowHand)

>         - 修复由于 PD 和 TiCDC 之间的网络隔离引起 TiCDC 程序异常退出的问题 [#8562](https://github.com/pingcap/tiflow/issues/8562) @[overvenus](https://github.com/overvenus)

Reviewer: ```suggestion
        - Fix the abnormal exit issue of the TiCDC service caused by network isolation between PD and TiCDC [#8562](https://github.com/pingcap/tiflow/issues/8562) @[overvenus](https://github.com/overvenus)
```

        - 修复在同步没有主键且包含非空唯一索引的表时，数据可能丢失的问题 [#8420](https://github.com/pingcap/tiflow/issues/8420) @[zhaoxinyu](https://github.com/zhaoxinyu)

>         - 修复 `db sorter` 使用内存时未受 `cgroup memory limit` 限制的问题 [#8588](https://github.com/pingcap/tiflow/issues/8588) @[amyangfei](https://github.com/amyangfei)
>         - 优化 `cdc cli` 在遇到非法输入时的错误提示 [#7903](https://github.com/pingcap/tiflow/issues/7903) @[CharlesCheung96](https://github.com/CharlesCheung96)

Reviewer: ```suggestion
        - Fix the issue that the memory usage of `db sorter` is not controlled by `cgroup memory limit` [#8588](https://github.com/pingcap/tiflow/issues/8588) @[amyangfei](https://github.com/amyangfei)
        - Optimize error messages of `cdc cli` for illegal input [#7903](https://github.com/pingcap/tiflow/issues/7903) @[CharlesCheung96](https://github.com/CharlesCheung96)
```

>         (dup: release-6.5.1.md > Bug fixes> Tools> TiCDC)- Fix the data inconsistency that occurs when replicating data to a TiDB or MySQL sink and when `CHARACTER SET` is specified on the column that has the non-null unique index without a primary key [#8420](https://github.com/pingcap/tiflow/issues/8420) @[zhaoxinyu](https://github.com/zhaoxinyu)

Reviewer: ```suggestion
        - (dup: release-6.5.1.md > Bug fixes> Tools> TiCDC)- Fix the data inconsistency that occurs when replicating data to a TiDB or MySQL sink and when `CHARACTER SET` is specified on the column that has the non-null unique index without a primary key [#8420](https://github.com/pingcap/tiflow/issues/8420) @[zhaoxinyu](https://github.com/zhaoxinyu)
```

>         - (dup): release-6.6.0.md > Bug 修复> Tools> TiCDC - 修复 redo log 容忍 S3 存储故障的时间过短的问题 [#8089](https://github.com/pingcap/tiflow/issues/8089) @[CharlesCheung96](https://github.com/CharlesCheung96)
>         - (dup): release-6.5.1.md > Bug 修复> Tools> TiCDC - 修复在 PD 异常时，暂停一个 changefeed 会错误设置状态的问题 [#8330](https://github.com/pingcap/tiflow/issues/8330) @[sdojjy](https://github.com/sdojjy)

Reviewer: ```suggestion
        - (dup): release-6.6.0.md > Bug fixes> Tools> TiCDC - Fix the issue of insufficient duration that redo log can tolerate for S3 storage failure [#8089](https://github.com/pingcap/tiflow/issues/8089) @[CharlesCheung96](https://github.com/CharlesCheung96)
        - (dup): release-6.5.1.md > Bug 修复> Tools> TiCDC - 修复在 PD 异常时，暂停一个 changefeed 会错误设置状态的问题 [#8330](https://github.com/pingcap/tiflow/issues/8330) @[sdojjy](https://github.com/sdojjy)
```

---
title: TiDB 6.1.6 Release Notes
summary: Learn about the compatibility changes, improvements, and bug fixes in TiDB 6.1.6.
---

# TiDB 6.1.6 Release Notes

Release date: April 12, 2023

TiDB version: 6.1.6

Quick access: [Quick start](https://docs.pingcap.com/tidb/v6.1/quick-start-with-tidb) | [Production deployment](https://docs.pingcap.com/tidb/v6.1/production-deployment-using-tiup) | [Installation packages](https://www.pingcap.com/download/?version=v6.1.6#version-list)

## Compatibility changes

- TiCDC fixes the issue of incorrect encoding of FLOAT data in Avro [#8490](https://github.com/pingcap/tiflow/issues/8490) @[3AceShowHand](https://github.com/3AceShowHand)

>     When upgrading the TiCDC cluster to v6.1.6 or later versions of v6.1.x, if a table replicated using Avro contains the `FLOAT` data type, you need to manually adjust the compatibility policy of Confluent Schema Registry to `None` before upgrading so that the changefeed can successfully update the schema. Otherwise, after upgrading, the changefeed will be unable to update the schema and enter an error state.

Author: ```suggestion
    When upgrading the TiCDC cluster to v6.1.6 or a later v6.1.x version, if a table replicated using Avro contains the `FLOAT` data type, you need to manually adjust the compatibility policy of Confluent Schema Registry to `None` before upgrading so that the changefeed can successfully update the schema. Otherwise, after upgrading, the changefeed will be unable to update the schema and enter an error state.
```
